lib/src/client/options.dart

```diff
@@ -93,15 +97,18 @@ class ChannelCredentials {
 class ChannelOptions {
   final ChannelCredentials credentials;
   final Duration idleTimeout;
+  final TransportType transportType;
   final BackoffStrategy backoffStrategy;
 
   const ChannelOptions(
       {ChannelCredentials credentials,
       Duration idleTimeout,
+      TransportType transportType = TransportType.Http2,
       BackoffStrategy backoffStrategy =
           defaultBackoffStrategy}) // Remove when dart-lang/sdk#31066 is fixed.
       : this.credentials = credentials ?? const ChannelCredentials.secure(),
         this.idleTimeout = idleTimeout ?? defaultIdleTimeout,
+        this.transportType = transportType ?? transportType,
```

Comment:
Yeah, that's weird. Will fix.

Reply:
So interesting dart related question -- technically even with default parameters, a caller could pass in "null" for any of these parameters.  Is it better to be safe and check for null or just use default parameters? Is there a "notnull" meta annotation (required is the closest I can find but not quite right).

Reply:
That's right, if you explicitly set a named parameter to null, it'll override the default value. There isn't a @notNull annotation available for public use--there is one that I believe can only be accessed internally by core Dart libraries.

Ultimately, I think it's up to the specific API owner to determine whether or not to allow null values, depending on what's going on. Generally though setting null to signify some special case or absent is a code smell that can be resolved by using [Optional<T>](https://github.com/google/quiver-dart/blob/master/lib/src/core/optional.dart) instead. But yeah, when I know something needs to be set I'll use [checkNotNull](https://github.com/google/quiver-dart/blob/54f6693e842c607ec4c9346c5003b4ce8f9e6c4d/lib/check.dart#L61) in the constructor.

It could very be (though unlikely) that the original author used parameter defaults to specify nullable fields and the initializers via '??' for non-nullable ones, though if that were the case it should have been documented. For TransportType in particular, I feel like we should enforce that it's set, since just blindly guessing a default value for Flutter or web will be wrong a lot.

Vaguely related: I do know that the built_value library goes the other way and uses  a "nullable" [annotation](https://github.com/google/built_value.dart/blob/c5ea1df7ae73bf3550604e5ab3c3b23b00cdce04/built_value/lib/built_value.dart#L90), though that's for a very specific type of data class.